Realm/RLMSyncConfiguration.h

```diff
@@ -85,6 +85,15 @@ NS_ASSUME_NONNULL_BEGIN
 */
 @property (nonatomic, nullable, copy) NSString *urlPrefix;
 
+/**
+ Whether nonfatal connection errors should cancel async opens.
+
+ By default, if a nonfatal connection error such as a connection timing out occurs, any currently pending asyncOpen operations will ignore the error and continue to retry until it succeeds. If this is set to true, the open will instead fail and report the error.
+
+  FIXME: This should probably be true by default in the next major version.
+ */
+@property (nonatomic) bool cancelAsyncOpenOnNonFatalErrors;
```

Comment:
Async open timeouts are implemented in Java/JS with a language-specific timer that aborts the sync session manually. This allows for a pr. Realm timeout when using Async Open. But one of the reasons was also that I didn't have any idea how to make that work effectively in ObjectStore/Sync. 

If you want to use what is currently available in OS/Sync, then I guess this makes sense but hopefully we can improve it in a future release.